soda_machine.rb

```diff
@@ -7,12 +9,31 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    @sodas.length
   end
 
   def find_soda(soda_brand)
+    @sodas.each do |soda|
+      if soda.brand == soda_brand
+        return soda
+      end
+    end
   end
 
   def sell(soda_brand)
+    s_soda = find_soda(soda_brand)
+    @sodas.delete(s_soda)
+    @cash += s_soda.price
+    return s_soda
```

Comment:
This ties in with my above question. The specs are expecting a nil value if no soda is found in both of these methods, but is that what your methods are returning?